core/src/main/java/io/questdb/network/Epoll.java

```diff
@@ -34,19 +34,22 @@
 
 public final class Epoll implements Closeable {
     private static final Log LOG = LogFactory.getLog(Epoll.class);
+
     private final int capacity;
     private final EpollFacade epf;
-    private final long epollFd;
+    private final int epollFd;
     private final long events;
     private long _rPtr;
     private boolean closed = false;
 
     public Epoll(EpollFacade epf, int capacity) {
         this.epf = epf;
         this.capacity = capacity;
-        this.events = _rPtr = Unsafe.calloc(EpollAccessor.SIZEOF_EVENT * (long) capacity, MemoryTag.NATIVE_IO_DISPATCHER_RSS);
-        // todo: this can be unsuccessful
+        this.events = this._rPtr = Unsafe.calloc(EpollAccessor.SIZEOF_EVENT * (long) capacity, MemoryTag.NATIVE_IO_DISPATCHER_RSS);
         this.epollFd = epf.epollCreate();
+        if (epollFd < 0) {
+            throw NetworkError.instance(epf.errno(), "cannot create epoll");
```

Comment:
"could not create epoll"

Reply:
Changed in bbec37baf832db9a846d04e59ab124f8bb84eadc